rocketpy/utilities.py

```diff
@@ -1,506 +1,622 @@
+    It is advised to use ";" as separator, but "," should work on most of cases.
+    The "," separator might cause problems if the data set contains lists where
+    the items are separated by commas.
+
+    Parameters
+    ----------
+    filename : string
+        String with the path to the .csv file. The file should follow the
+        following structure:
+
+            attribute_class; parameter_name; mean_value; standard_deviation;
+            environment; ensembleMember; [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];;
+            motor; impulse; 1415.15; 35.3;
+            motor; burnOut; 5.274; 1;
+            motor; nozzleRadius; 0.021642; 0.0005;
+            motor; throatRadius; 0.008; 0.0005;
+            motor; grainSeparation; 0.006; 0.001;
+            motor; grainDensity; 1707; 50;
+
+    Returns
+    -------
+    dictionary
+        Dictionary with all rocket data to be used in dispersion analysis. The
+        dictionary will follow the following structure:
+            analysis_parameters = {
+                'environment': {
+                    'ensembleMember': [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+                },
+                'motor': {
+                    'impulse': (1415.15, 35.3),
+                    'burnOut': (5.274, 1),
+                    'nozzleRadius': (0.021642, 0.0005),
+                    'throatRadius': (0.008, 0.0005),
+                    'grainSeparation': (0.006, 0.001),
+                    'grainDensity': (1707, 50),
+                    }
+            }
+    """
+    try:
+        file = np.genfromtxt(
+            filename, usecols=(1, 2, 3), skip_header=1, delimiter=";", dtype=str
+        )
+    except ValueError:
+        warnings.warn(
+            f"Error caught: the recommended delimiter is ';'. If using ',' instead, be "
+            + "aware that some resources might not work as expected if your data "
+            + "set contains lists where the items are separated by commas. "
+            + "Please consider changing the delimiter to ';' if that is the case."
+        )
+        warnings.warn(traceback.format_exc())
+        file = np.genfromtxt(
+            filename, usecols=(1, 2, 3), skip_header=1, delimiter=",", dtype=str
+        )
+    analysis_parameters = dict()
+    for row in file:
+        if row[0] != "":
+            if row[2] == "":
+                try:
+                    analysis_parameters[row[0].strip()] = float(row[1])
+                except ValueError:
+                    analysis_parameters[row[0].strip()] = eval(row[1])
+            else:
+                try:
+                    analysis_parameters[row[0].strip()] = (float(row[1]), float(row[2]))
+                except ValueError:
+                    analysis_parameters[row[0].strip()] = ""
+    return analysis_parameters
+
+
+def apogee_by_mass(flight, min_mass=3, max_mass=30, points=10):
+    """Returns a Function object that estimates the apogee of a rocket given
```

Comment:
The function could have a `plot=False` optional argument. This would avoid having to define a variable when calling this function and then doing `variable.plot` to actually see the plot